assets/src/edit-story/app/media/utils/useTranscodeVideo.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { v4 as uuidv4 } from 'uuid';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../config';
+
+const isDevelopment = process.env.NODE_ENV === 'development';
+
+const getFileName = ({ name }) => name.split('.').slice(0, -1).join('.');
+
+function useTranscodeVideo() {
+  const { ffmpegCoreUrl } = useConfig();
+  const isFeatureEnabled = useFeature('videoOptimization');
+
+  /**
+   * Transcode a video using ffmpeg.
+   *
+   * @param {File} file File object of original video file.
+   * @return {Promise<File>} File object of transcoded video file.
+   */
+  async function transcodeVideo(file) {
+    const { createFFmpeg, fetchFile } = await import(
+      /* webpackChunkName: "ffmpeg" */ '@ffmpeg/ffmpeg'
+    );
+    const ffmpeg = createFFmpeg({
+      corePath: ffmpegCoreUrl,
+      log: isDevelopment,
+    });
+    await ffmpeg.load();
+
+    const tempFileName = uuidv4() + '.mp4';
```

Comment:
Should mp4 be a const?

Reply:
I think in #5993 the desired output file type will become a function parameter instead.